huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_batch_publishment.go

```diff
@@ -0,0 +1,174 @@
+						"biz_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+							Description: `Specifies the ID of the object to be published.`,
+						},
+						"biz_type": {
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+							Description: `Specifies the type of the object to be published.`,
+						},
+					},
+				},
+			},
+			"approver_user_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the user ID of the architecture reviewer.",
+			},
+			"approver_user_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the user name of the architecture reviewer.",
+			},
+			"fast_approval": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Whether to automatically review.",
+			},
+		},
+	}
+}
+
+func resourceArchitectureBatchPublishmentCreate(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/design/approvals/batch-publish"
+		product = "dataarts"
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"workspace":    d.Get("workspace_id").(string),
+			"Content-Type": "application/json;charset=UTF-8",
+		},
+		JSONBody: utils.RemoveNil(buildCreateArchitectureBatchPublishmentBodyParams(d)),
+	}
+
+	createResp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error publishing DataArts Architecture resource: %s", err)
+	}
+
+	createRespBody, err := utils.FlattenResponse(createResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	groupId := utils.PathSearch("data.value.group_id", createRespBody, nil)
+	if groupId == nil {
+		return diag.Errorf("error creating DataArts Architecture batch publishment: ID is not found in API response")
+	}
+
+	publishStatus := utils.PathSearch("data.value.operation_status", createRespBody, nil)
+	errMsg := ""
+	if publishStatus == "FAILED" {
+		for _, v := range utils.PathSearch("data.value.groups", createRespBody, make([]interface{}, 0)).([]interface{}) {
+			if utils.PathSearch("operation_status", v, nil) != "FAILED" {
+				continue
+			}
+			bizId := utils.PathSearch("biz_id", v, "").(string)
+			failedMsg := utils.PathSearch("remark", v, "").(string)
+			errMsg += fmt.Sprintf("%s | %s;\n", bizId, failedMsg)
+		}
+	}
+
+	if errMsg != "" {
+		return diag.Errorf("error publishing some resources: %s", errMsg)
+	}
```

Comment:
```suggestion
	errMsgs := make([]string, 0)
	if publishStatus == "FAILED" {
		for _, v := range utils.PathSearch("data.value.groups", createRespBody, make([]interface{}, 0)).([]interface{}) {
			if utils.PathSearch("operation_status", v, nil) != "FAILED" {
				continue
			}
			bizId := utils.PathSearch("biz_id", v, "").(string)
			failedMsg := utils.PathSearch("remark", v, "").(string)
			errMsgs = append(errMsgs, fmt.Sprintf("%s | %s;\n", bizId, failedMsg))
		}
	}

	if errMsg != "" {
		return diag.Errorf("error publishing some resources: %s", strings.Split(errMsgs, ";"))
	}
```
how about uses the character ';' to split the messages.

Reply:
fixed